server/db/driver/pg/orders.go

```diff
@@ -325,6 +325,37 @@ func (a *Archiver) BookOrders(base, quote uint32) ([]*order.LimitOrder, error) {
 	return limits, nil
 }
 
+// EpochOrders retrieves all epoch orders for the specified market.
+func (a *Archiver) EpochOrders(base, quote uint32) ([]*order.LimitOrder, []*order.MarketOrder, error) {
+	marketSchema, err := a.marketSchema(base, quote)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	tableName := fullOrderTableName(a.dbName, marketSchema, true) // active (true)
+
+	// no query timeout here, only explicit cancellation
+	ords, err := ordersByStatusFromTable(a.ctx, a.db, tableName, base, quote, orderStatusEpoch)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Verify loaded order type and add to correct slice.
+	limits := []*order.LimitOrder{}
+	markets := []*order.MarketOrder{}
+	for _, ord := range ords {
+		switch o := ord.(type) {
+		case *order.LimitOrder:
+			limits = append(limits, o)
+		case *order.MarketOrder:
+			markets = append(markets, o)
+		default:
+			log.Errorf("loaded epoch order %v that was not a limit or market order: %T", ord.ID(), ord)
```

Comment:
Epoch status cancel orders are a thing too though.

Reply:
I can't think of a **good** way of getting market specific cancel orders. Can I add base/quote to the cancel orders table?

Reply:
I mean, the query is loading them.  Just pointing out this error will happen.

Reply:
Oh, does `ordersByStatusFromTable` dump them first?  Hmmm

Reply:
Srry! I had made a stupid comment you are probably replying to and the deleted it. I was thinking the cancel orders aren't market specific, but they also have market specific table names. I was confused but ok now.

Reply:
Err, maybe I am confused. Cancels are in a separate table correct?

Reply:
I *think* it's correct now.

Reply:
Ah I was thinking of matches, which are in the same table.  You're right that cancels have their own table.